_LowLevelCode/cpp/serialiser/c_deserialize.cpp

```diff
@@ -309,10 +309,15 @@ mxArray* deserialize(uint8_t* data, size_t& memPtr, size_t size, bool recursed)
     case STRUCT:
     {
         uint32_t nFields;
-
-        deser(data, memPtr, &nFields, types_size[UINT32]);
-        std::vector<uint32_t> fNameLens(nFields);
-        deser(data, memPtr, fNameLens, nFields * types_size[UINT32]);
+        std::vector<uint32_t> fNameLens(0);
+        if ((nDims == 2) && (memPtr >= size)) {
```

Comment:
I've added my guess why it is necessary but this is just guess.  This change fixes intefimttent crash in deserialization, and chasing it properly is a separate pretty difficult tast. But change here is necessary, as I would be committing broken code otherwise.